### PR TITLE
chore: Suggest opening a new issue in Feedback form

### LINKED ISSUE #512: Suggest opening a new issue in Feedback form
Our latest received feedbacks were more like feature requests.

We should point users to open a new github issue in these cases so it's easier for them and us to keep track of the progress.

Current feedback form:

image

### PR DESCRIPTION
**Pull request by luis-herasme**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Suggest opening a new issue in Feedback form

### Related Issue (Optional)
Closes #512

### Screenshots:
Before:
image
image
image

After:
image
image
image

Before:
image

After:
image

### CODE REVIEW

File: apps/web/src/components/FeedbackWidget.tsx [hunk 1]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+            🙂
+          </Emoji>
+        </div>
+        <Button className="mt-4 w-full" onClick={sendFeedback}>
+          Submit
+        </Button>
+        <hr className="mt-4 border-border-light dark:border-border-dark" />
+        <p className="mt-4">
+          We also recommend that you open a new GitHub issue so it&#39;s easier
+          for you and us to keep track of the progress.
```
Comment (PabloCastellano) [inline apps/web/src/components/FeedbackWidget.tsx:161]: I suggest the following wording:

Please [open a new issue](https://github.com/Blobscan/blobscan/issues) if you have a feature request or want to report a bug.

**Review by PabloCastellano — COMMENTED** [review]
I liked the old styles better

**Review by xFJA — COMMENTED** [review]
Good job! Left a few comments! I have some topics to discuss:
- Should we use a `form` here? We could wrap the component in a `<form>`
- I think we could create a folder for `FeedbackWidget` and create `FeedbackCard` and `Emoji` in different files (similar to `Dropdown` implementation). This way we improve the readability of the component

File: apps/web/src/components/FeedbackWidget.tsx (hunk 1 above)
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [e0016d9](https://github.com/Blobscan/blobscan/pull/602/commits/e0016d9c000b3215927e887ed1bf334d4b172122)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 2]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+        >
+          Feedback <Chat className="h-5 w-5" />
+        </Button>
+      </div>
+      <FeedbackCard open={open} close={() => setOpen(false)} />
+    </>
   );
 };
+
+function FeedbackCard({ open, close }: { open: boolean; close: () => void }) {
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [e610d49](https://github.com/Blobscan/blobscan/pull/602/commits/e610d4959a5db9197f02102dee81da3959aacb31)

[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [9d5145a](https://github.com/Blobscan/blobscan/pull/602/commits/9d5145a0ee5963e3898579ae7bf3a18e40c569e3)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 3]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+          <Link href="https://github.com/Blobscan/blobscan/issues/new">
+            Open a new issue on GitHub
+          </Link>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function Emoji({
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [c79b316](https://github.com/Blobscan/blobscan/pull/602/commits/c79b3163feaf3b8f3f3aad1ac98c6baeeb723c1b)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 4]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+  children,
+  active,
+  onClick,
+}: {
+  children: string;
+  active: boolean;
+  onClick: () => void;
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx:181]: Done [c79b316](https://github.com/Blobscan/blobscan/pull/602/commits/c79b3163feaf3b8f3f3aad1ac98c6baeeb723c1b)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 5]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+        <Button className="mt-4 w-full" onClick={sendFeedback}>
+          Submit
+        </Button>
+        <hr className="mt-4 border-border-light dark:border-border-dark" />
+        <p className="mt-4">
+          We also recommend that you open a new GitHub issue so it&#39;s easier
+          for you and us to keep track of the progress.
+        </p>
+        <div className="mt-4">
+          <Link href="https://github.com/Blobscan/blobscan/issues/new">
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [439fe23](https://github.com/Blobscan/blobscan/pull/602/commits/439fe233a8358b0a7f52d81bc29f1b22aee89499)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 6]
```diff
@@ -24,22 +27,173 @@ export const FeedbackWidget: React.FC = function () {
...
+        <div className="mt-4 flex items-center justify-center gap-4 text-3xl">
+          <Emoji
+            active={emoji === "🙁"}
+            onClick={() => {
+              if (emoji === "🙁") {
+                setEmoji("");
+              } else {
+                setEmoji("🙁");
+              }
+            }}
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx]: Done [5eb67f5](https://github.com/Blobscan/blobscan/pull/602/commits/5eb67f57c0f6a943410c181bbccb48668a173db1)

File: apps/web/src/components/FeedbackWidget.tsx [hunk 7]
```diff
@@ -1,11 +1,14 @@
...
+import Chat from "~/icons/chat.svg";
+import Close from "~/icons/close.svg";
+import { Button } from "./Button";
+import { IconButton } from "./IconButton";
+import { Link } from "./Link";
 
 export const FeedbackWidget: React.FC = function () {
-  const { pathname, query } = useRouter();
-  const { resolvedTheme } = useTheme();
+  const [open, setOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget.tsx:12]: When using primitive types, we normally do not add the type:
https://github.com/Blobscan/blobscan/blob/ba1d78376100a65389cc4a9eac9e399ff2f3627d/apps/web/src/hooks/useClickOutside.ts#L11

File: apps/web/src/pages/api/feedback/enabled.ts [hunk 8]
```diff
@@ -3,5 +3,5 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { env } from "~/env.mjs";
 
 export default async function (req: NextApiRequest, res: NextApiResponse) {
-  return res.status(200).json({ enabled: !!env.FEEDBACK_WEBHOOK_URL });
+  return res.status(200).json(Boolean(env.FEEDBACK_WEBHOOK_URL));
```
Comment (PJColombo) [inline apps/web/src/pages/api/feedback/enabled.ts:6]: Can we revert this change?

I'd like to keep consistency with the rest of the API and always return a JSON

**Review by PJColombo — CHANGES_REQUESTED** [review]
Nice job!

I left some comments.

Also, can we optimise the svgs sizes by running `pnpm svg:format` from the root of the monorepo?

File: apps/web/src/components/FeedbackWidget/Emoji.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  function onClick() {
+    if (active) {
+      setEmoji("");
+    } else {
+      setEmoji(emoji);
+    }
+  }
+
+  return (
+    <button
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget/Emoji.tsx]: Done [a1705b0](https://github.com/Blobscan/blobscan/pull/602/commits/a1705b03917c664a0bb3982aa6b01fc696208470)

File: apps/web/src/pages/api/feedback/enabled.ts (hunk 8 above)
Comment (luis-herasme) [reply apps/web/src/pages/api/feedback/enabled.ts]: Done [70dc1c9](https://github.com/Blobscan/blobscan/pull/602/commits/70dc1c99e6afb726f579ffb28de02bc810a096db)

File: apps/web/src/components/FeedbackWidget/Emoji.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import type { FC } from "react";
+
+interface EmojiProps {
+  emoji: string;
+  currentEmoji: string;
+  setEmoji: (emoji: string) => void;
```
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/components/FeedbackWidget/Emoji.tsx]: Done [c44c263](https://github.com/Blobscan/blobscan/pull/602/commits/c44c263afda3717902bae6c5ce646bde649ed88b)

**Review by xFJA — APPROVED** [review]

**Review by PJColombo — APPROVED** [review]
